src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -431,9 +433,106 @@ public SlaveTemplate(
         this.metadataHopsLimit =
                 metadataHopsLimit != null ? metadataHopsLimit : EC2AbstractSlave.DEFAULT_METADATA_HOPS_LIMIT;
         this.enclaveEnabled = enclaveEnabled != null ? enclaveEnabled : EC2AbstractSlave.DEFAULT_ENCLAVE_ENABLED;
+        this.associateIPStrategy = associateIPStrategy != null ? associateIPStrategy : AssociateIPStrategy.DEFAULT;
```

Comment:
If the `associateIPStrategy` is `null`, shouldn't we check want to use `AssociateIPStrategy.backwardsCompatible(associatePublicIp)` instead of the `AssociateIPStrategy.DEFAULT` ?